File: Utilities/Correlation_validation_script.m
% McGrattan
% March 21, 2011
% Correlation_validation_script.m
%
% This script creates the plots that are included in the Correlation Guide.
% It consists of calls to other scripts contained within the
% subfolder called "scripts". 
%
% The most important script is called dataplot. It reads the file called
% Correlation_validation_dataplot_inputs.csv and generates 1000+ plots. If you
% want to process only some of these plots, comment out the other 
% scripts and change the data plot line as follows:
%
% [saved_data,drange] = dataplot(Dataplot_Inputs_File,Validation_Dir,Manuals_Dir,[a:b]);
%
% where a and b are the lines in the .csv file you want to process.
% Alternatively, you can specify the "Dataname" you want:
%
% [saved_data,drange] = dataplot(Dataplot_Inputs_File,Validation_Dir,Manuals_Dir,'WTC');
%
% In this case, all the WTC results will be plotted.
%
% Dataplot creates most of the plots for the Validation Guide.
% It must be run before scatplot, which makes the scatter plots.

close all
clear all

script_dir = [pwd, '/../../fds/Utilities/Matlab/scripts/'];
addpath(script_dir)

% Dataplot and scatplot options

Dataplot_Inputs_File = [pwd, '/Correlation_validation_dataplot_inputs.csv'];
EXP_Dir = [pwd, '/../../exp/'];
OUT_Dir = [pwd, '/../Validation/'];
Manuals_Dir = [pwd, '/../Manuals/'];
Scatterplot_Inputs_File = [pwd, '/Correlation_validation_scatterplot_inputs.csv'];
Scatterplot_Dir = [pwd, '/../../Manuals/SCRIPT_FIGURES/ScatterPlots/'];

% Statistics output options

Stats_Output = 'Validation';
Output_File = [pwd, '/Correlation_validation_scatterplot_output.csv'];
Statistics_Tex_Output = [pwd, '/../Manuals/SCRIPT_FIGURES/Scatterplots/validation_statistics.tex'];
Histogram_Tex_Output = [pwd, '/../Manuals/SCRIPT_FIGURES/Scatterplots/validation_histograms.tex'];

% Run dataplot and scatplot scripts
[saved_data,drange] = dataplot(Dataplot_Inputs_File, EXP_Dir, OUT_Dir, Manuals_Dir);
scatplot(saved_data, drange, ...
         'Manuals_Dir', Manuals_Dir, ...
         'Scatterplot_Inputs_File', Scatterplot_Inputs_File, ...
         'Stats_Output', Stats_Output, ...
         'Scatterplot_Dir', Scatterplot_Dir, ...
         'Output_File', Output_File, ...
         'Statistics_Tex_Output', Statistics_Tex_Output, ...
         'Histogram_Tex_Output', Histogram_Tex_Output)
 
display('validation scripts completed successfully!')

File: Utilities/Correlation_validation_script.py
#!$FIREMODELS/fds/.github/fds_python_env/bin/python

import subprocess
import fdsplotlib
import matplotlib.pyplot as plt
import importlib
import runpy
importlib.reload(fdsplotlib) # use for development (while making changes to fdsplotlib.py)
print("Using:", fdsplotlib.__file__)

# Dataplot and scatplot options

Dataplot_Inputs_File = 'Correlation_validation_dataplot_inputs.csv'
EXP_Dir = '../../exp/'
OUT_Dir = '../Validation/'
Manuals_Dir = '../Manuals/'
Scatterplot_Inputs_File = 'Correlation_validation_scatterplot_inputs.csv'
Scatterplot_Dir = '../Manuals/SCRIPT_FIGURES/Scatterplots/'

# Statistics output options

Stats_Output = 'Validation'

# Run dataplot and scatplot scripts

saved_data, drange = fdsplotlib.dataplot(config_filename=Dataplot_Inputs_File,
                                         expdir=EXP_Dir,
                                         cmpdir=OUT_Dir,
                                         pltdir=Manuals_Dir,
                                         close_figs=True,
                                         verbose=True,
                                         plot_range=['all'],
                                         ) # see notes below on plot_range

# ----- write saved_data, drange to disk -----
import pickle

# Save
with open("saved_data_validation.pkl", "wb") as f:
    pickle.dump((saved_data, drange), f)

# Later...
with open("saved_data_validation.pkl", "rb") as f:
    saved_data, drange = pickle.load(f)
#---------------------------------------------

fdsplotlib.scatplot(saved_data,drange,
                    Manuals_Dir=Manuals_Dir,
                    Scatterplot_Inputs_File=Scatterplot_Inputs_File,
                    Stats_Output=Stats_Output,
                    Scatterplot_Dir=Scatterplot_Dir,
                    verbose=True,
                    )

print("Python validation scripts completed successfully!")

# ------------------------------
# plot_range usage examples
#
# plot_range lets you select which rows of the config file to process.
# You can mix row numbers, ranges, and Dataname strings:
#
#  1. Single row by number (Spreadsheet-style, including header rows):
#       plot_range = [1995]
#
#  2. Inclusive ranges by "start:stop":
#       plot_range = ["5:9"]        # rows 5 through 9
#
#  3. Open-ended ranges:
#       plot_range = ["1995:"]      # from row 1995 to the end
#
#  4. Named selection by Dataname (case-insensitive):
#       plot_range = ["CSTB Tunnel", "Steckler Compartment"]
#
#  5. Mixed selection:
#       plot_range = [1, 2, "5:9", "CSTB Tunnel", "7000:"]
#
#  6. All rows:
#       plot_range = ["all"]
#
# Notes:
# - Row numbers are 1-based (like Spreadsheet).
# - Ranges are inclusive, e.g. "5:9" means 5,6,7,8,9.
# - "start:" runs to the last row.
# - Strings that are not ranges or "all" are matched to the Dataname column.
# ------------------------------

